wagtail/admin/viewsets/model.py

```diff
@@ -102,6 +115,25 @@ def delete_view(self):
             header_icon=self.icon,
         )
 
+    @cached_property
+    def menu_label(self):
+        return self.model_opts.verbose_name_plural.title()
```

Comment:
Will add tests for this and the default `name` in a separate PR when we focus on `ModelViewSet`. For now, the changes in this PR are mostly focused on the base `ViewSet` class, so I didn't add any `ModelViewSet`-specific tests.